agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/AzureMonitorMetricExporter.java

```diff
@@ -0,0 +1,109 @@
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.exporter;
+
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.DOUBLE_GAUGE;
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.DOUBLE_SUM;
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.HISTOGRAM;
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.LONG_GAUGE;
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.LONG_SUM;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.builders.MetricTelemetryBuilder;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.TelemetryItem;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.FormattedTime;
+import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AzureMonitorMetricExporter implements MetricExporter {
+
+  private static final List<String> EXCLUDED_METRIC_NAMES = new ArrayList<>();
+
+  private final TelemetryClient telemetryClient;
+  private static final Logger logger = LoggerFactory.getLogger(AzureMonitorMetricExporter.class);
+  private final AtomicBoolean stopped = new AtomicBoolean();
+
+  public AzureMonitorMetricExporter(TelemetryClient telemetryClient) {
+    this.telemetryClient = telemetryClient;
+  }
+
+  static {
+    EXCLUDED_METRIC_NAMES.add("http.server.active_requests");
+    EXCLUDED_METRIC_NAMES.add("http.server.duration");
```

Comment:
let's exclude also:

* `http.client.duration`
* `rpc.client.duration`
* `rpc.server.duration`

I'm a little concerned about keeping up-to-date with these. Can we add validation in all of our smoke tests to ensure that extra "custom" metrics are not received?

Reply:
smoke tests took care of it.. otherwise it won't pass. it expected the right metrics. size won't match if custom metrics show up. where will these custom metrics go then? I need more context.

Reply:
rpc.client.duration and rpc.server.duration should get generated in the gRPC smoke test

http.client.duration should get generated in HttpClient smoke test

Reply:
what I was thinking, is if we can add validation to _all_ smoke tests, that way if otel starts generating messaging metrics, that would cause our JMS and Kafka smoke tests to fail and we would know to add those to the exclusion list

Reply:
that can be a separate pr. will exclude those three.